Tests/TensorFlowTests/FreezableTests.swift

```diff
@@ -0,0 +1,74 @@
+// Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import XCTest
+@testable import TensorFlow
+
+final class FreezableTests: XCTestCase {
+    func testFreezableParameters() {
+        struct FreezableDense : Layer {
+            @Freezable var weight: Float
+            // Workaround for underlying storage private access.
+            var freezableWeight: Freezable<Float> {
+                get { $weight }
+                set { $weight = newValue }
+            }
+
+            @Freezable var bias: Float
+            // Workaround for underlying storage private access.
+            var freezableBias: Freezable<Float> {
+                get { $bias }
+                set { $bias = newValue }
+            }
+
+            @differentiable
+            func callAsFunction(_ input: Float) -> Float {
+                return input * weight + bias
+            }
+        }
+
+        var dense = FreezableDense(weight: 2, bias: 3)
+        let x: Float = 4
+        do {
+            let (value, gradient) = valueWithGradient(at: dense, x) { dense, x in dense(x) }
+            XCTAssertEqual(11, value)
+            // FIXME: '$' is not a valid identifier:
+            //     cannot declare entity named '$weight'; the '$' prefix is reserved for
+            //     implicitly-synthesized declarations.
+            //
+            // Tentative solution: change `Differentiable` derived conformances to use original
+            // property names instead of underlying storage property names.
+            //
+            // Now: `FreezableDense.TangentVector($weight: ..., $bias: ...)
+            // Goal: `FreezableDense.TangentVector(weight: ..., bias: ...)
+            //
+            // XCTAssertEqual(FreezableDense.TangentVector($weight: 4, $bias: 1), gradient.0)
+            XCTAssertEqual(4, gradient.0.$weight)
+            XCTAssertEqual(1, gradient.0.$bias)
+            XCTAssertEqual(2, gradient.1)
+        }
+        dense.freezableWeight.isFrozen = true
```

Comment:
Why is `dense.$weight.isFrozen = true` not working?

Reply:
It doesn't work because underlying storage is currently always private:
```
/Users/danielzheng/swift-apis/Tests/TensorFlowTests/FreezableTests.swift:62:15: error: '$weight' is inaccessible due to 'private' protection level
        dense.$weight.isFrozen = true
              ^~~~~~~
/Users/danielzheng/swift-apis/Tests/TensorFlowTests/FreezableTests.swift:21:28: note: '$weight' declared here
            @Freezable var weight: Float
                           ^
```

Indicated with a comment:
```swift
struct FreezableDense : Layer {
    @Freezable var weight: Float
    // Workaround for underlying storage private access.
    var freezableWeight: Freezable<Float> {
        get { $weight }
        set { $weight = newValue }
    }
}
```

The ability to specify underlying storage access levels is important for accessing APIs defined on property wrapper types. @tanner0101 made a case for customizing underlying storage access levels in [this post](https://forums.swift.org/t/se-0258-property-wrappers-second-review/25843/14?u=dan-zheng).

Reply:
This indicates that the feature is not quite ready for our use case yet. We should review the 2nd iteration of the proposal and make sure things are going the right direction for us.

Reply:
Are both underlying properties and functions private? For example, could we define a `freeze()`/`unfreeze()` function in the property wrapper?

Reply:
The issue is that wrapper value is `private`, or `internal` at most.

Reply:
According to @DougGregor's recent [post](https://forums.swift.org/t/se-0258-property-wrappers-second-review/25843/117?u=tanner0101), this may be getting changed:

> Here's the set of things I want to revise:
> ...
> When wrapperValue is present, the $foo property becomes available and has the same access level as the original property foo, so when A has a wrapperValue (say it returns a Ref<Value>).

If so, that should make property wrappers viable for your use case. 👍

Reply:
Yeah, that'd be great!